testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py

```diff
@@ -3896,6 +3897,115 @@ def test_blob_contructor_replication(params_from_base_test_setup, blob_data_type
+    sg_url = params_from_base_test_setup["sg_url"]
+    base_url = params_from_base_test_setup["base_url"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    db = params_from_base_test_setup["db"]
+    db_config = params_from_base_test_setup["db_config"]
+
+    # Reset cluster to ensure no data in system
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    # 1. create 100 docs on SGW
+    sg_db = "db"
+    num_sg_docs = 100
+    channels = ["ABC"]
+    username = "autotest"
+    password = "password"
+
+    sg_client = MobileRestClient()
+    sg_client.create_user(sg_admin_url, sg_db, username, password=password, channels=channels)
+    cookie, session_id = sg_client.create_session(sg_admin_url, sg_db, username)
+    auth_session = cookie, session_id
+    sg_added_docs = sg_client.add_docs(url=sg_url, db=sg_db, number=num_sg_docs, id_prefix="sg_doc", channels=channels, auth=auth_session, attachments_generator=attachment_generator)
+
+    # 2a. delete 10 docs in SGW
+    docs_to_delete_1 = sg_added_docs[0:10]
+    sg_client.delete_docs(url=sg_url, db=sg_db, docs=docs_to_delete_1, auth=auth_session)
+    # 2b. verify the 10 deleted docs are in tombstone state
+    sg_docs = sg_client.get_all_docs(url=sg_admin_url, db=sg_db, include_docs=True)["rows"]
+    assert len(sg_docs) == 90, "Expected number of docs is incorrect after doc deletion in sync-gateway"
+    for sg_doc in sg_docs:
+        assert sg_doc not in docs_to_delete_1, "Deleted doc should not show on sg doc list"
+
+    # 3a. create a cbl db, but do not create any doc (cbl db is empty)
+    cbl_db_name = "empty-db-" + str(time.time())
+    cbl_db = db.create(cbl_db_name, db_config)
+    # 3b. alternate scenario - create docs on cbl then delete all docs on cbl
+    if cbl_action_before_init_replication:
+        db.create_bulk_docs(db=cbl_db, number=9, id_prefix="pre-repl", channels=channels)
+        cbl_doc_ids = db.getDocIds(cbl_db)
+        db.delete_bulk_docs(cbl_db, cbl_doc_ids)
+
+    # 4. create a pull replicator, start replication
+    replicator = Replication(base_url)
+    authenticator = Authenticator(base_url)
+    replicator_authenticator = authenticator.authentication(session_id, cookie, authentication_type="session")
+    repl = replicator.configure_and_replicate(source_db=cbl_db,
+                                              target_url=sg_blip_url,
+                                              continuous=False,
+                                              replicator_authenticator=replicator_authenticator,
+                                              replication_type="pull")
+
+    # 5. verify SGW stats, cbl_replication_pull.rev_send_count sent to CBL equals to 90
+    expvars = sg_client.get_expvars(sg_admin_url)
+    rev_send_count = expvars["syncgateway"]["per_db"][sg_db]["cbl_replication_pull"]["rev_send_count"]
+    if cbl_action_before_init_replication:
```

Comment:
did not get this condition verification as it is missing in summary. Before starting the replication, 10 docs deleted in SGW , and only 90 docs left.  I though it will be rev_send_count is 100 as it is pull replicator. Not sure if I misunderstood